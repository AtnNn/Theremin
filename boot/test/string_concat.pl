:- string_concat(X,b,ab), print(X), nl.
:- string_concat(a,X,ab), print(X), nl.
:- string_concat(a,b,X), print(X), nl.
:- string_concat(a,"b","ab"), print(yes), nl; print(no), nl.
:- string_concat(a,"b","abc"), print(yes), nl; print(no), nl.

a --> "foo", "bar"; "baz".

:- a(X, ""), print(X), nl, fail.
