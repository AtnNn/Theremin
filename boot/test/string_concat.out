"a"
"b"
.(a, b)
yes
no
.("foo", .("bar", ""))
.("baz", "")
